### ISSUE TITLE
Not tree-shakeable

### ISSUE THREAD

**Issue by exoego** [supercharge/promise-pool#83]
I use [esbuild](https://esbuild.github.io/) to bundle my app which uses promise-pool like this:

```typescript
import { PromisePool } from "@supercharge/promise-pool"
...
new PromisePool()
        .withConcurrency(this.parallelism)
        .for(...)
        .process(...)
```

I noticed that unused code `return-value` and `contracts` are bundled in the artifact.
It is due to re-exporting in https://github.com/supercharge/promise-pool/blob/bf79b02cd4f0612dd4730d8c2946a3ec62e1288c/src/index.ts#L7-L12
and promise-pool is currently published as CommonJS, not ESM.

So it would be if this library [declare `"sideEffects": false` in package.json ](https://webpack.js.org/guides/tree-shaking/#mark-the-file-as-side-effect-free)or publish ESM version.

**Comment by marcuspoehls** [comment]
@exoego Hey, I’ve published version `3.2.0` containing the `sideEffects: false` change in package.json 🙂

**Comment by exoego** [comment]
Thanks!
